rollup/internal/controller/watcher/batch_proposer.go

```diff
@@ -155,6 +155,28 @@ func (p *BatchProposer) TryProposeBatch() {
 }
 
 func (p *BatchProposer) updateDBBatchInfo(batch *encoding.Batch, codecVersion encoding.CodecVersion, metrics utils.BatchMetrics) error {
+	for {
+		compatible, err := utils.CheckBatchCompressedDataCompatibility(batch, codecVersion)
+		if err != nil {
+			log.Error("Failed to check batch compressed data compatibility", "codecVersion", codecVersion, "err", err)
+			return err
+		}
+
+		if compatible {
+			break
+		}
+
+		if len(batch.Chunks) == 1 {
+			log.Error("Cannot truncate batch with only 1 chunk for compatibility", "start block number", batch.Chunks[0].Blocks[0].Header.Number.Uint64(),
+				"end block number", batch.Chunks[0].Blocks[len(batch.Chunks[0].Blocks)-1].Header.Number.Uint64())
+			return errors.New("cannot truncate batch with only 1 chunk for compatibility")
+		}
+
+		log.Info("Batch not compatible with compressed data, removing last chunk")
```

Comment:
I recommend addint more info here, maybe the batch index and the number of chunks?

Reply:
added more info in [85f995b](https://github.com/scroll-tech/scroll/pull/1431/commits/85f995ba36832342801724b92fe0aaa06b180646).